File: src/test/java/com/hiddewieringa/elevator/domain/elevator/model/PersonAggregateTest.java
package com.hiddewieringa.elevator.domain.elevator.model;

import com.hiddewieringa.elevator.domain.elevator.ElevatorCreated;
import com.hiddewieringa.elevator.domain.person.*;
import com.hiddewieringa.elevator.domain.person.model.PersonAggregate;
import com.hiddewieringa.elevator.domain.person.model.PersonId;
import org.axonframework.test.aggregate.AggregateTestFixture;
import org.axonframework.test.aggregate.FixtureConfiguration;
import org.junit.Before;
import org.junit.Test;

import java.util.UUID;

public class PersonAggregateTest {

    private FixtureConfiguration<PersonAggregate> fixture;

    @Before
    public void setUp() {
        fixture = new AggregateTestFixture<>(PersonAggregate.class);
    }

    @Test
    public void create() {
        PersonId personId = new PersonId(UUID.randomUUID());
        fixture.given()
                .when(new PersonArrives(personId, 0, 1))
                .expectEvents(new PersonArrived(personId, 0, 1));
    }

    @Test
    public void enter() {
        PersonId personId = new PersonId(UUID.randomUUID());
        fixture.given(new PersonArrived(personId, 0, 1))
                .when(new EnterElevator(personId))
                .expectEvents(new PersonEnteredElevator(personId));
    }

    @Test
    public void enter2() {
        PersonId personId = new PersonId(UUID.randomUUID());
        fixture.given(new PersonArrived(personId, 0, 1),
                new PersonEnteredElevator(personId))
                .when(new EnterElevator(personId))
                .expectException(IllegalArgumentException.class);
    }

    @Test
    public void enterLeave() {
        PersonId personId = new PersonId(UUID.randomUUID());
        fixture.given(new PersonArrived(personId, 0, 1),
                new PersonEnteredElevator(personId))
                .when(new LeaveElevator(personId))
                .expectEvents(new PersonLeftElevator(personId));
    }

    @Test
    public void leave2() {
        PersonId personId = new PersonId(UUID.randomUUID());
        fixture.given(new PersonArrived(personId, 0, 1))
                .when(new LeaveElevator(personId))
                .expectException(IllegalArgumentException.class);
    }
}

File: src/test/kotlin/com/hiddewieringa/elevator/domain/elevator/model/PersonAggregateTest.kt
package com.hiddewieringa.elevator.domain.elevator.model

import com.hiddewieringa.elevator.domain.person.*
import com.hiddewieringa.elevator.domain.person.model.PersonAggregate
import com.hiddewieringa.elevator.domain.person.model.PersonId
import org.axonframework.test.aggregate.AggregateTestFixture
import org.axonframework.test.aggregate.FixtureConfiguration
import org.junit.Before
import org.junit.Test
import java.util.*

class PersonAggregateTest {

    private lateinit var fixture: FixtureConfiguration<PersonAggregate>

    @Before
    fun setUp() {
        fixture = AggregateTestFixture(PersonAggregate::class.java)
    }

    @Test
    fun create() {
        val personId = PersonId(UUID.randomUUID())
        fixture.given()
                .`when`(PersonArrives(personId, 0, 1))
                .expectEvents(PersonArrived(personId, 0, 1))
    }

    @Test
    fun enter() {
        val personId = PersonId(UUID.randomUUID())
        fixture.given(PersonArrived(personId, 0, 1))
                .`when`(EnterElevator(personId))
                .expectEvents(PersonEnteredElevator(personId))
    }

    @Test
    fun enter2() {
        val personId = PersonId(UUID.randomUUID())
        fixture.given(PersonArrived(personId, 0, 1),
                PersonEnteredElevator(personId))
                .`when`(EnterElevator(personId))
                .expectException(IllegalArgumentException::class.java)
    }

    @Test
    fun enterLeave() {
        val personId = PersonId(UUID.randomUUID())
        fixture.given(PersonArrived(personId, 0, 1),
                PersonEnteredElevator(personId))
                .`when`(LeaveElevator(personId))
                .expectEvents(PersonLeftElevator(personId))
    }

    @Test
    fun leave2() {
        val personId = PersonId(UUID.randomUUID())
        fixture.given(PersonArrived(personId, 0, 1))
                .`when`(LeaveElevator(personId))
                .expectException(IllegalArgumentException::class.java)
    }
}
